### PR TITLE
Initial plugin implementation

### PR DESCRIPTION
**Pull request by christophermaier**
Largely taken from our existing release candidate code from
https://github.com/grapl-security/buildkite-common, but repackaged as
a proper Buildkite plugin.

Signed-off-by: Christopher Maier <chris@graplsecurity.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,108 @@
...
+non-surprising. This makes it easy to identify the _project_ without
+having to add extra plugin configuration, and without having to create
+some non-standard Pulumi reference, like `project/stack`.)
+
+## Building
+
+Requires `make`, `docker`, and `docker-compose`.
+
+Running `make` will run all formatting, linting, and testing.
+>>>>>>> 0137228... Initial plugin implementation
```
Comment (jgrillo-grapl) [inline README.md:108]: looks like something left over from a resolved merge conflict

File: lib/rc_test.sh [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    if [ -f "${ALL_COMMANDS}" ]; then
+        cat "${ALL_COMMANDS}"
+    fi
+}
+
+oneTimeSetUp() {
+    export BUILDKITE_BUILD_URL="https://buildkite.com/grapl/pipeline-infrastructure-verify/builds/2112"
+    export ALL_COMMANDS="${SHUNIT_TMPDIR}/all_commands"
+    export GIT_AUTHOR_NAME="Testy McTestface"
+    export GIT_AUTHOR_EMAIL="tests@tests.com"
```
Comment (colin-grapl) [inline lib/rc_test.sh:49]: I'm assuming this won't come up or be an issue, but `tests.com` is a real site. Maybe we should use `example.com` here?

**Review by jgrillo-grapl — APPROVED** [review]

File: lib/rc_test.sh (hunk 2 above)
Comment (christophermaier) [reply lib/rc_test.sh:49]: Ah, good point. It won't be an issue here, but it's probably not a bad habit to start using `example.com` more broadly.

File: README.md (hunk 1 above)
Comment (christophermaier) [reply README.md:108]: Aaaargh... well spotted.

File: hooks/command [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# Read contents of the artifact file
+########################################################################
+contents=$(artifact_contents "${artifact_file_name}")
+echo "--- Retrieved artifact metadata from '${artifact_file_name}'"
+echo "${contents}"
+
+# STOLEN FROM https://github.com/buildkite-plugins/docker-buildkite-plugin/blob/9f90d8ef742d9fa1eb3556720e16f2b842ff1cb2/hooks/command#L25-L47
+#
+# Reads a list from plugin config into a global result array
+# Returns success if values were read
```
Comment (wimax-grapl) [inline hooks/command:31]: gross, but such is bash

**Review by wimax-grapl — APPROVED** [review]

**Pull request merged**